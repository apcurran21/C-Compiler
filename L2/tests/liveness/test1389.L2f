(@fibo
	1
	%n <- rdi
	cjump %n <= 2 :leq_2
	rdi -= 2
	mem rsp -8 <- :callfibo_1_finish
	call @fibo 1
	:callfibo_1_finish
	%res <- rax
	rdi <- %n
	rdi -= 1
	mem rsp -8 <- :callfibo_2_finish
	call @fibo 1
	:callfibo_2_finish
	%res += rax
	goto :done
	:leq_2
	cjump %n <= 0 :less_0
	%res <- 1
	goto :done
	:less_0
	%res <- 0
	:done
	rax <- %res
	return
)
