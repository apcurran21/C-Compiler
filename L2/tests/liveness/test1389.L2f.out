(
(in
(r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rax rbp rbx)
(%n r12 r13 r14 r15 rax rbp rbx)
(%n r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rbp rbx rdi %res)
(r12 r13 r14 r15 rbp rbx rdi %res)
(r12 r13 r14 r15 rbp rbx rdi %res)
(r12 r13 r14 r15 rax rbp rbx %res)
(r12 r13 r14 r15 rax rbp rbx %res)
(r12 r13 r14 r15 rbp rbx %res)
(%n r12 r13 r14 r15 rbp rbx)
(%n r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%n r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rbp rbx rdi)
(%n r12 r13 r14 r15 rax rbp rbx)
(%n r12 r13 r14 r15 rax rbp rbx)
(%n r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rbp rbx rdi %res)
(r12 r13 r14 r15 rbp rbx rdi %res)
(r12 r13 r14 r15 rbp rbx rdi %res)
(r12 r13 r14 r15 rax rbp rbx %res)
(r12 r13 r14 r15 rax rbp rbx %res)
(r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rbp rbx %res)
(%n r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rbp rbx %res)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
